flow/project.py

```diff
@@ -2411,208 +2388,114 @@ def _fetch_status(
-            results[index[label_result["job_id"]]]["_labels_error"] = label_result[
-                "_labels_error"
-            ]
-
-        return results
-
-    def _fetch_status_in_parallel(
-        self, pool, jobs, groups, ignore_errors, cached_status
-    ):
-        try:
-            serialized_project = cloudpickle.dumps(self)
-            serialized_tasks_labels = [
-                (
-                    cloudpickle.loads,
-                    serialized_project,
-                    job.get_id(),
-                    ignore_errors,
-                    "fetch_labels",
-                )
-                for job in jobs
-            ]
-            serialized_tasks_groups = [
-                (
-                    cloudpickle.loads,
-                    serialized_project,
-                    group,
-                    ignore_errors,
-                    cached_status,
-                    "fetch_status",
+        with self._buffered():
+            aggregate_groups = list(
+                self._generate_selected_aggregate_groups_with_status(
+                    scheduler_info=scheduler_info,
+                    selected_aggregates=aggregates,
+                    selected_groups=single_operation_groups,
                 )
-                for group in groups
-            ]
-        except Exception as error:  # Masking all errors since they must be pickling related.
-            raise self._PickleError(error)
+            )
+            status_results = parallel_executor(
+                compute_status,
+                aggregate_groups,
+                desc="Fetching status",
+                file=err,
+            )
 
-        label_results = pool.starmap(_serializer, serialized_tasks_labels)
-        group_results = pool.starmap(_serializer, serialized_tasks_groups)
+        with self._buffered():
+            compute_labels = functools.partial(
+                self._get_job_labels,
+                ignore_errors=ignore_errors,
+            )
+            job_labels = parallel_executor(
+                compute_labels,
+                distinct_jobs,
+                desc="Fetching labels",
+                file=err,
+            )
 
-        return label_results, group_results
+        status_results_combined = []
+        for aggregate_id, aggregate_results in groupby(
+            sorted(status_results, key=lambda result: result["aggregate_id"]),
+            key=lambda result: result["aggregate_id"],
+        ):
+            status_results_combined.append(
+                {
+                    "aggregate_id": aggregate_id,
+                    "groups": {
+                        result["group_name"]: result["status"]
+                        for result in aggregate_results
+                    },
+                    # TODO: fix operations error propagation (bdice is not
+                    # sure what that data is supposed to look like)
+                    "_error": None,
```

Comment:
Is the issue here that we'd need to group the error messages from various groups in some consistent fashion?

Reply:
Ah, I forgot about this TODO. This needs to be handled. If errors occur, that information is propagated through the status somehow. I need to figure out what the "error" data structure should look like and group it, as you indicated.

Reply:
This is resolved now.